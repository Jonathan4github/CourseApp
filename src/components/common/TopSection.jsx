import React from 'react';
import PropTypes from 'prop-types';


const propTypes = {
  pageTitle: PropTypes.string.isRequired,
  buttonLabel: PropTypes.string.isRequired,
  redirectFunc: PropTypes.func.isRequired
};

const TopSection = ({ pageTitle, buttonLabel, redirectFunc }) => (
  <div className="top-container">
    <h2>{pageTitle}</h2>
    <button
      type="button"
      className="btn btn-outline-primary add-course-button btn-sm"
      onClick={redirectFunc}>
      {buttonLabel}
    </button>
  </div>
);

TopSection.propTypes = propTypes;

export default TopSection;
