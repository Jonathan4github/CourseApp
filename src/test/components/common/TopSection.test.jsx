/* eslint-disable */
import TopSection from '../../../components/common/TopSection';

const setup = () => {
  const props = {
    pageTitle: '',
    buttonLabel: '',
    redirectFunc: jest.fn()
  };

  return mount(<TopSection {...props} />);
};

let mountWrapper;

describe('<TopSection /> component test', () => {
  it('renders component without crashing', () => {
    mountWrapper = setup();
    expect(toJson(mountWrapper)).toMatchSnapshot();
  });

  it('renders appropriate elements', () => {
    mountWrapper = setup();
    expect(mountWrapper.find('h2').length).toEqual(1);
    expect(mountWrapper.find('button').length).toEqual(1);
  });
});
