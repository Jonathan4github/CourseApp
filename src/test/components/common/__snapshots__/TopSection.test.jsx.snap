// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TopSection /> component test renders component without crashing 1`] = `
<TopSection
  buttonLabel=""
  pageTitle=""
  redirectFunc={[MockFunction]}
>
  <div
    className="top-container"
  >
    <h2 />
    <button
      className="btn btn-outline-primary add-course-button btn-sm"
      onClick={[MockFunction]}
      type="button"
    />
  </div>
</TopSection>
`;
